### PR TITLE
Feature/navgation body builder

### LINKED ISSUE #76: NavigationBody builder
**Is your feature request related to a problem? Please describe.**
To reduce memory consumption, we should be able to lazily build navigation body items

**Describe the solution you'd like**
Like ListView.builder, provide an index based builder to NavigationBody

**Describe alternatives you've considered**
I considered (and am using) a prebuild list of widgets instead of a builder

### PR DESCRIPTION
**Pull request by NicolaVerbeeck**
This adds a builder variant to NavigationBody. Fixes #76

## Pre-launch Checklist

- [X] I have run `dartfmt` on all changed files  
- [X] I have updated `CHANGELOG.md` with my changes  
- [X] I have run "optimize/organize imports" on all changed files
- [X] I have addressed all analyzer warnings as best I could
- [X] I have added/updated relevant documentation
- [X] I have run `flutter pub publish --dry-run` and addressed any warnings

### CODE REVIEW

**Comment by bdlukaa** [comment]
Shouldn't we be using `Listview.builder` when itembuilder not null? That would increase performance

**Comment by NicolaVerbeeck** [comment]
I don't think so, we are using a building pattern anyway in the class by only building the widget at the current index. A listview builder does not make sense here

File: lib/src/controls/navigation/navigation_view/body.dart [hunk 1]
```diff
@@ -13,18 +13,45 @@ class NavigationBody extends StatefulWidget {
   const NavigationBody({
     Key? key,
     required this.index,
-    required this.children,
+    required List<Widget> children,
     this.transitionBuilder,
     this.animationCurve,
     this.animationDuration,
   })  : assert(index >= 0 && index <= children.length),
+        // ignore: prefer_initializing_formals
```
Comment (bdlukaa) [inline lib/src/controls/navigation/navigation_view/body.dart:21]: use ignore for file for both

**Review by bdlukaa — APPROVED** [review]

File: lib/src/controls/navigation/navigation_view/body.dart (hunk 1 above)
Comment (NicolaVerbeeck) [reply lib/src/controls/navigation/navigation_view/body.dart]: Done

**Pull request merged**